tensornetwork/network_operations.py

```diff
@@ -734,11 +735,12 @@ def contract_trace_edges(node: AbstractNode) -> AbstractNode:
   Raises:
     ValueError: If `node` has no trace edges.
   """
-  for edge in node.edges:
+  res = node
+  for edge in res.edges:
     if edge.is_trace():
-      return contract_parallel(edge)
-  raise ValueError('`node` has no trace edges')
-
+      res = contract_parallel(edge)
+      break
+  return res
```

Comment:
Will this no longer raise a ValueError if `node` has no trace edges? Update docstring if so.